src/main/js/helpers/axis.js

```diff
@@ -205,26 +221,67 @@ const getAxesScale = (axis, scale, config) => {
             config.axis.y2.domain.upperLimit - config.axis.y2.domain.lowerLimit
         );
 
+    if (
+        config.suppressTrailingZeros &&
+        utils.isUndefined(config.axis.y.ticks.format)
+    ) {
+        axis.y = d3.axisLeft(scale.y);
+        tickFormatToTrimTrailingZeros = tickFormatter(axis.y);
+    }
+
     axis.y = prepareYAxis(
         scale.y,
         config.axis.y.domain,
         config.height,
-        getAxisTickFormat(config.locale, config.axis.y.ticks.format),
+        getAxisTickFormat(
+            config.locale,
+            utils.isDefined(config.axis.y.ticks.format)
+                ? config.axis.y.ticks.format
+                : tickFormatToTrimTrailingZeros
+        ),
         config.ticksCount
     );
 
     if (hasY2Axis(config.axis)) {
+        if (
+            config.suppressTrailingZeros &&
+            utils.isUndefined(config.axis.y2.ticks.format)
+        ) {
+            axis.y2 = d3.axisRight(scale.y2);
+            tickFormatToTrimTrailingZeros = tickFormatter(axis.y2);
+        }
+
         axis.y2 = prepareY2Axis(
             scale.y2,
             config.axis.y2.domain,
             config.height,
-            getAxisTickFormat(config.locale, config.axis.y2.ticks.format),
+            getAxisTickFormat(
+                config.locale,
+                utils.isDefined(config.axis.y2.ticks.format)
+                    ? config.axis.y2.ticks.format
+                    : tickFormatToTrimTrailingZeros
+            ),
             config.ticksCount
         );
     }
     return axis;
 };
 
+/**
+ * Inserts '~' just before the format type to suppress ticks values's trailing zeros when default d3 tick formatting is used
+ *
+ * @private
+ * @param {object} axis - Scaled axes object
+ * @returns {string} tick format string
+ */
+const tickFormatter = (axis) => {
+    const defaultTickFormat = axis.scale().tickFormat().toString();
+    return `${defaultTickFormat.slice(
+        0,
+        defaultTickFormat.length - 1
+    )}~${defaultTickFormat.slice(defaultTickFormat.length - 1)}`;
```

Comment:
Can you add a comment here what exactly we are returning? and provide me one example of return value for this method?

Reply:
Added comment [here](https://github.com/JananiGunasekaran/carbon-graphs/commit/655b2444b4ec1bee262ef03c4e59f429f3f9b6cd). 

According to d3-format documentation [here](https://github.com/d3/d3-format/blob/master/README.md#locale_format), the ~ character will trim the insignificant trailing zeros.
So we get the default d3 tick format and insert '~' just before the format type.

I have console logged the default d3 tick format and the value returned by this tickFormatter () method here:
![image](https://user-images.githubusercontent.com/60426528/85978933-42fd9180-b9fd-11ea-8ed1-025cf7f78e2a.png)